stream_alert_cli/terraform/handler.py

```diff
@@ -151,40 +146,43 @@ def terraform_handler(options, config):
 
 
 def terraform_build(options, config):
-    """Run Terraform with an optional set of targets
+    """Run Terraform with an optional set of targets and clusters
 
     Args:
         options (namedtuple): Parsed arguments from manage.py
     """
-    # Generate Terraform files
     if not terraform_generate(config=config):
         return
-    # Target is for terraforming a specific streamalert module.
-    # This value is passed as a list
-    if options.target == ['athena']:
-        tf_runner(targets=['module.stream_alert_athena'])
-    elif options.target == ['threat_intel_downloader']:
-        tf_runner(targets=['module.threat_intel_downloader'])
-    elif options.target:
-        targets = ['module.{}_{}'.format(target, cluster)
-                   for cluster in config.clusters()
-                   for target in options.target]
-        tf_runner(targets=targets)
-    else:
-        tf_runner()
+
+    # Define the set of custom targets to apply
+    tf_runner_targets = set()
+    # If resource are not clustered, it is most likely required to
+    # fall in the custom mapping below:
+    custom_module_mapping = {
+        'athena': 'module.stream_alert_athena',
+        'threat_intel_downloader': 'module.threat_intel_downloader'
+    }
+    clusters = set(options.cluster or config.clusters())
+
+    if options.target:
+        tf_runner_targets.update({
+            'module.{}_{}'.format(target, cluster)
+            for cluster in clusters for target in options.target
+        })
+        for name, _ in custom_module_mapping.iteritems():
```

Comment:
if you don't need to use the value of in this for loop, don't use `.iteritems` and instead just do `for name in custom_module_mapping:...`